flexprep/domain/db_utils.py

```diff
@@ -60,43 +62,144 @@ def insert_item(self, item: IFSForecast) -> None:
+                # Query for step-0 count
+                cursor = self.conn.execute(step_zero_count_query, (forecast_ref_time,))
+                step_zero_count = cursor.fetchone()[0]
+
+                # If fewer than two step-0 items exist, return an empty list
+                if step_zero_count < 2:
+                    logger.info(
+                        f"Currently only {step_zero_count} step=0 files are available. "
+                        "Waiting for these before processing."
+                    )
+                    return []
+
+                # Proceed with querying the steps
+                tstart = CONFIG.main.time_settings.tstart
+                tincr = CONFIG.main.time_settings.tincr
+
+                query = (
+                    "SELECT row_id, forecast_ref_time, step, key, processed "
+                    "FROM uploaded "
+                    "WHERE forecast_ref_time = ? "
+                    "AND processed = FALSE "
+                    "AND step != 0 "
+                    "AND (step - ?) % ? = 0 "
+                    "ORDER BY step ASC"
+                )
+
+                cursor = self.conn.execute(query, (forecast_ref_time, tstart, tincr))
                 rows = cursor.fetchall()
-                logger.info(f"Query returned {len(rows)} items.")
-                results = [
-                    IFSForecast(
-                        row_id=row[0],
-                        forecast_ref_time=row[1],
-                        step=row[2],
-                        key=row[3],
-                        processed=row[4],
+
+                logger.info(f"Query returned {len(rows)} pending timestep(s).")
+
+                # Prepare the combined list of processable steps
+                combined_steps = []
+
+                for row in rows:
+                    current_step = row[2]  # The 'step' field
+                    previous_step = current_step - tincr
+
+                    # Get two step-0 items
+                    step_zero_query = (
+                        "SELECT * FROM uploaded "
+                        "WHERE forecast_ref_time = ? AND step = 0 "
+                        "LIMIT 2"
+                    )
+                    cursor = self.conn.execute(step_zero_query, (forecast_ref_time,))
+                    step_zero_items = cursor.fetchall()
+
+                    # Create step-0 IFSForecast objects
+                    step_zero_forecasts = [
+                        IFSForecast(
+                            row_id=zero_row[0],
+                            forecast_ref_time=dt.strptime(
+                                zero_row[1], "%Y-%m-%d %H:%M:%S"
+                            ),
+                            step=int(zero_row[2]),
+                            key=zero_row[3],
+                            processed=zero_row[4],
+                        ).to_dict()
+                        for zero_row in step_zero_items
+                    ]
```

Comment:
This part could be taken out of the loop and `processable_list` could be initialised on a copy of `step_zero_forecasts`